201-front-door-managed-waf-ruleset/azuredeploy.json

```diff
@@ -20,59 +20,50 @@
                 "description": "The name of the WAF policy"
             }
         },
-        "ruleGroupOverride": {
+        "wafMode": {
             "type": "string",
             "allowedValues": [
-                "SqlInjection",
-                "XSS"
+                "Detection",
+                "Prevention"
             ],
+            "defaultValue": "Detection",
             "metadata": {
-                "description": "The azure managed provider override configuration."
-            }
-        },
-        "ruleGroupOverrideAction": {
-            "type": "string",
-            "allowedValues": [
-                "Allow",
-                "Block",
-                "Log"
-            ],
-            "metadata": {
-                "description": "Type of Action to perform. Must be Allow, Block or Log"
+                "description": "Describes if it is in detection mode or prevention mode at policy level."
             }
         }
     },
     "variables": {
-        "frontdoorref": "[resourceId('Microsoft.Network/frontDoors', parameters('frontDoorName'))]",
-        "wafpolicyref": "[resourceId('Microsoft.Network/FrontDoorWebApplicationFirewallPolicies', parameters('wafPolicyName'))]",
-        "frontdoorLocation": "global"
+        "frontdoorLocation": "global",
+        "frontdoorSuffix": ".azurefd.net"
```

Comment:
this needs to be retrieved with the reference() function

Reply:
The resource cannot make a reference to itself while is being created.